Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -69,27 +63,101 @@ public override void Initialize()
+                continue;
+
+            var headRevCount = _antagSelection.CalculateAntagCount(ev.Players.Length, comp.PlayersPerHeadRev, comp.MaxHeadRevs);
+
+            var headRevs = _antagSelection.ChooseAntags(headRevCount, eligiblePlayers);
+
+            _antagSelection.SendBriefing(headRevs, Loc.GetString("head-rev-role-greeting"), Color.CornflowerBlue, comp.HeadRevStartSound);
+            GiveHeadRev(headRevs, comp.HeadRevPrototypeId, comp);
+        }
     }
 
     /// <summary>
-    /// Checks if the round should end and also checks who has a mindshield.
+    /// Called when a Head Rev uses a flash in melee to convert somebody else.
     /// </summary>
-    protected override void ActiveTick(EntityUid uid, RevolutionaryRuleComponent component, GameRuleComponent gameRule, float frameTime)
+    private void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref AfterFlashedEvent ev)
     {
-        base.ActiveTick(uid, component, gameRule, frameTime);
-        if (component.CommandCheck <= _timing.CurTime)
+        TryComp<AlwaysRevolutionaryConvertibleComponent>(ev.Target, out var alwaysConvertibleComp);
+        var alwaysConvertible = alwaysConvertibleComp != null;
+
+        if (!_mind.TryGetMind(ev.Target, out var mindId, out var mind) && !alwaysConvertible)
+            return;
+
+        if (HasComp<RevolutionaryComponent>(ev.Target) ||
+            HasComp<MindShieldComponent>(ev.Target) ||
+            !HasComp<HumanoidAppearanceComponent>(ev.Target) &&
+            !alwaysConvertible ||
+            !_mobState.IsAlive(ev.Target) ||
+            HasComp<ZombieComponent>(ev.Target))
         {
-            component.CommandCheck = _timing.CurTime + component.TimerWait;
+            return;
+        }
+
+        _npcFaction.AddFaction(ev.Target, RevolutionaryNpcFaction);
+        EnsureComp<RevolutionaryComponent>(ev.Target);
+        _stun.TryParalyze(ev.Target, comp.StunTime, true);
+        if (ev.User != null)
+        {
+            _adminLogManager.Add(LogType.Mind, LogImpact.Medium, $"{ToPrettyString(ev.User.Value)} converted {ToPrettyString(ev.Target)} into a Revolutionary");
+
+            if (_mind.TryGetRole<RevolutionaryRoleComponent>(ev.User.Value, out var headrev))
+                headrev.ConvertedCount++;
 
-            if (CheckCommandLose())
-            {
-                _roundEnd.DoRoundEndBehavior(RoundEndBehavior.ShuttleCall, component.ShuttleCallTime);
-                GameTicker.EndGameRule(uid, gameRule);
-            }
+        }
+
+        if (mindId == default || !_role.MindHasRole<RevolutionaryRoleComponent>(mindId))
+        {
+            _role.MindAddRole(mindId, new RevolutionaryRoleComponent { PrototypeId = RevolutionaryAntagRole });
+        }
+
+        if (mind != null && mind.OwnedEntity.HasValue)
+            _antagSelection.SendBriefing(mind.OwnedEntity.Value, Loc.GetString("rev-role-greeting"), Color.Red, new SoundPathSpecifier("/Audio/Ambience/Antag/headrev_start.ogg"));
```

Comment:
Sound path specifiers shouldnt be made like this, move it to a datafield in the component

Reply:
I can see why this was done, there is no easy access to RevolutionaryRuleComponent in this method.
Should i convert the SoundSpecifier to a static field?
Otherwise i will need to either duplicate the SoundSpecifier on RevolutionaryComponent or query GameRules whenever someone is flashed.

Reply:
yeah just repeat the field, not that big of a deal (tho i would assume the RevolutionaryComponent would be accessible whenever `RevolutionaryRuleComponent.HeadRevStartSound` is read, if so just move it from the rule. Havent checked thom could be wrong)